src/lib/extensions.ts

```diff
@@ -0,0 +1,27 @@
+import { OptionValues } from 'commander'
+import isEmpty from 'lodash/isEmpty.js'
+import BaseCommand from '../commands/base-command.js'
+import { init } from '../commands/init/init.js'
+import { log } from '../utils/command-helpers.js'
+
+export const packagesThatNeedSites = ['@netlify/neon']
+
+export async function handleExtensionRequirements(options: OptionValues, command: BaseCommand) {
+  const { project } = command
+  const { site, siteInfo } = command.netlify
+
+  const packageJson = await project.getPackageJSON()
+  const dependencies = packageJson.dependencies ?? {}
```

Comment:
perf nit: move this below into `if (!hasSiteData)` to avoid this I/O in 99.9% of cases